source/Octopus.Client/Repositories/BasicRepository.cs

```diff
@@ -47,13 +47,22 @@ protected virtual void CheckSpaceResource(IHaveSpaceResource spaceResource)
                 {
                     var spaceRoot = Repository.LoadSpaceRootDocument();
                     var isDefaultSpaceFound = spaceRoot != null;
-
-                    if (!isDefaultSpaceFound)
+                  
+                    if (!isDefaultSpaceFound && CheckServerVersionIncludesSpaces(Repository.LoadRootDocument().Version))
                     {
                         throw new DefaultSpaceNotFoundException(spaceResource);
                     }
                 });
         }
+        
+        private bool CheckServerVersionIncludesSpaces(string version)
```

Comment:
While this implementation works, it might be better to do feature detection, rather than hard coding versions. We already do this over here: https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Client/OctopusAsyncRepository.cs#L295

Reply:
[nit] A better name for this function might be something like `ServerSupportsSpaces` or `IsSpacesSupported` or `IsSpacesSupportedByServer`. In particular, the `Check` part of the name isn't very helpful. The method name would be better if it only reflected the value we are trying to obtain.